vcd/mutexkv.go

```diff
@@ -43,9 +45,9 @@ func (m *MutexKV) get(key string) *sync.Mutex {
 	return mutex
 }
 
-// Returns a properly initalized MutexKV
-func NewMutexKV() *MutexKV {
-	return &MutexKV{
+// Returns a properly initalized mutexKV
+func newMutexKV() *mutexKV {
```

Comment:
Curious, why are we making these functions private?

Reply:
These functions were in SDK 1.x, but were removed in 2.0.
Hashicorp has released the code in the public domain and I added it to our code base. But since we don't need to export them, I've made them private